feat(docs): add gas price feedback documentation to the sequencing doc (#10725)
iotaledger/iota: docs/content/about-iota/iota-architecture/sequencing.mdx

The case is illustrated in the following image.

![Sequencing example, final sequencing](/img/about-iota/iota-architecture/sequencing2.svg)

## Gas Price Feedback for Cancelled Transactions

When a transaction is cancelled due to shared object congestion, IOTA provides
a **suggested gas price** in the execution status of that cancelled transaction.
This suggested gas price informs the sender about the minimum gas price they
should have paid for their transaction to have been successfully scheduled in
the _same_ consensus commit.

Other networks such as Sui only report which shared objects were congested, but
not what gas price would have been enough for successful execution. Users on
those networks have to rely on dry-run simulations to figure out a sufficient
gas price. IOTA instead computes the suggested gas price from the actual

> scheduling decisions in the commit, so the sender already knows what gas price

Reviewer: Maybe we should soften this sentence. Instead of saying that “the sender already knows what gas price to bid when retrying”, it might be more accurate to say that the protocol provides price feedback based on the actual scheduling decisions in the previous commit, which helps the sender estimate a sufficient gas price for a retry.

Author: Done

to bid when retrying.

### How the suggested gas price is calculated

> - The gas price feedback algorithm performs imaginary allocation of the

Reviewer: ```suggestion
- The gas price feedback algorithm simulates the allocation of the
```

cancelled transaction at the latest possible execution start time within the
commit's capacity, so that only the last scheduled transactions on each
congested object overlap with it.
- The clearing gas price for each object is the maximum gas price across those

> last scheduled transactions that overlap with the imaginary allocation of the

Reviewer: ```suggestion
last scheduled transactions that overlap with the simulated allocation of the
```

cancelled transaction.
- The suggested gas price is then calculated as the maximum clearing gas price

> across all input objects of the cancelled transaction, plus one - this ensures

Reviewer: ```suggestion
across all input objects of the cancelled transaction plus one, ensuring
```

> the retried transaction would outbid competitors and get scheduled.

Reviewer: ```suggestion
that it would outbid competing transactions and get scheduled.
```

- The suggested gas price value is capped at the protocol's maximum gas price.

> ### Using the suggested gas price

Reviewer: In this paragraph we can probably also mention the prediction. I can add a sentence about it in the other documentation PR.

Author: Sounds good! Thanks for the suggestions!
